### DISCUSSION TITLE
Control States "left" and "right" are swapped

### DISCUSSION THREAD

**Discussion by HumanDuck23** [PrismarineJS/mineflayer#2439] [Q&A]
Hi!
I'm making a proxy server with a built in bot using the proxy example in node-minecraft-protocol. Instead of creating a new client for `targetClient` like in the example, I create a bot and use its _client as the `targetClient`. This works, and to tell the vanilla client connecting to the proxy I send regular position packets. 

When the bot joins the server (hypixel in this case) I want it to walk forward to the NPC with the title "The Bridge". This part works fine with this code:

```js
for (let key of Object.keys(bot.entities)) {
    if (bot.entities[key].metadata[2] === "§bThe Bridge") {
        bot.lookAt(bot.entities[key].position, false, () => {
            console.log("looking, walking!")
            bot.setControlState("forward", true)
        })
    }
}
```
But when I add `bot.setControlState("left", true)` under the "forward", it starts moving to the right. In addition, the `bot.entity.yaw` and `bot.entity.pitch` seemed to be off, to send the correct values to the vanilla client I had to turn them from radians to degrees, invert the +/- and add 180 degrees. I made sure the bot is actually looking where I think it's looking by joining the lobby in a normal game and watching the bot move. Has anyone had a problem like this?

**Comment by rom1504** [comment]
hi

mineflayer code handles converting to/from "mojang degrees", are you using that?

**Reply by HumanDuck23** [reply]
ah, I must have missed that. I'll take a look later, thanks
